controllers/awsmachine_controller.go

```diff
@@ -473,6 +477,10 @@ func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *
 		}
 	}
 
+	if err := r.reconcileLBAttachment(machineScope, clusterScope, instance); err != nil {
```

Comment:
The way @randomvariable recently restructured this above is that we have two sections of operations: those applicable to all known states, and those only applicable to operational states. This moves LB attachment reconciliation outside of and after those two sections. Is that appropriate? Would it still work if we moved it inside the "all known states" section, or would that be too early?

Reply:
Yeah we talked about it, if the Machine is being stopped it's still an operational state, but the ELB should remove the Machine until it gets into a running state again

Reply:
The reasoning makes sense to me.

Reply:
If we move it into the "all known states" section, wouldn't that cover it (unless it's too early i.e. we need the security groups ensured first)?

Reply:
Rereading, I think it will depend on https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1710/files/82e0a992239c3f2ccba1038ea2d52c585c054a19#r420397258

Reply:
Which file/line? (It's not easy to tell what the link is pointing to.)

Reply:
I meant the comment above, for line 575 of `controllers/awsmachine_controller.go`:

> Is elbsvc.DeregisterInstanceFromAPIServerELB(i) allowed for stopping or stopped instances?
> If so, that's fine.
> Otherwise might need to filter those out given this is now not called within the operational states.

Reply:
I moved the code under "all known states"